Source/TrackableEntities.EF.5/DbContextExtensions.cs

```diff
@@ -199,16 +199,31 @@ public static void ApplyChanges(this InterceptorPool pool, IEnumerable<ITrackabl
                     context.ApplyChangesOnProperties(item, visitationHelper.Clone(), TrackingState.Deleted, interceptors);
                 }
 
+                var isComplex = context.IsComplexType(item.GetType());
+                bool isParentEligible(TrackingState parentState) => parentState == TrackingState.Modified || parentState == TrackingState.Unchanged;
+                var isModifiable = 
+                          (isComplex 
+                            ? isParentEligible(parent.TrackingState) 
+                            : (item.TrackingState == TrackingState.Modified && (state == null || state == TrackingState.Modified)))                          
+                          && item.ModifiedProperties?.Count > 0;
+
                 // Set modified properties
-                if (item.TrackingState == TrackingState.Modified
-                    && (state == null || state == TrackingState.Modified)
-                    && item.ModifiedProperties != null
-                    && item.ModifiedProperties.Count > 0)
+                if (isModifiable)
                 {
                     // Mark modified properties
                     SetEntityState(context, item, parent, propertyName, EntityState.Unchanged, interceptors);
                     foreach (var property in item.ModifiedProperties)
-                        context.Entry(item).Property(property).IsModified = true;
+                    {
+                        if(isComplex)
+                        {
+                            var parentEntry = context.Entry(parent);
+                            if(parentEntry.State == EntityState.Detached)
+                              parentEntry.State = parent.TrackingState.ToEntityState();
+                            parentEntry.ComplexProperty(propertyName).Property(property).IsModified = true;                           
+                        }
+                        else
+                            context.Entry(item).Property(property).IsModified = true;
+                    }
```

Comment:
I thought about moving the `context.Entry()` call outside the `for each` loop so we don't have to retrieve the entity again on each iteration.
The ugly thing here is that `parentEntry.ComplexProperty(propertyName)` returns a `DbComplexPropertyEntry` and `context.Entry(item)`, a `DbEntityEntry<ITrackable>` which are not assignable from each other, so the `Property()` function is not a shared one. Meh, could do it if I hadn't involved the complex properties. Apparently `ComplexProperties` do come with their tiny complications.